core/client/index.ts

```diff
@@ -101,12 +149,97 @@ export class Core {
   
   async onMessage(msg: inboundEncPayloadType) {
     console.log("seqID: " + msg.seqID);
-    let payload: string = this.olmWrapper.decrypt(
-        msg.encPayload,
-        msg.sender,
+    let decPayload: string = this.olmWrapper.decrypt(
+      msg.encPayload,
+      msg.sender,
     );
+
+    let validationPayload = null;
+    let messagePayload = null;
+    let recipients = null;
+    let consistencyLoopback = null;
+
+    try {
```

Comment:
Same with this whole block of new code. There should basically be one call to validate the incoming/account-for message.

Reply:
We discussed this and agree there perhaps should be one call from core to the message chain integration called `validate_and_insert` that handles a lot of the code you mentioned. It should also transparently add (and remove) required fields to the payload, however it is essential to keep the basic validate and insert operations separate since they operate on different message content (quite literally do not share any input parameters other than the fact payloads arrived in the same message and should be entirely independent of each other). However, there should be some way to return that an invalid message was detected so that a client can initiate another message to the client through which we detected the reordering so they can detect it too. We may want to execute insert message independently of whether the validation function succeeded or failed.

Reply:
>  it is essential

Do you mean internally, in the library? Sure... though I'm not sure why it would be "essential"

> However, there should be some way to return that an invalid message was detected so that a client can initiate another message to the client through which we detected the reordering so they can detect it too. 

I think this is asking too much at this stage. At the point we've detected a re-ordering, the server is a bad actor. How might we hope to use it to relay a message to our peer that the server is a bad actor? Until at least OSDI (at least), we don't need to do this, we just panic.